src/source/Common/AwsV4Signer.c

```diff
@@ -1070,3 +1024,100 @@ PCHAR getRequestVerbString(HTTP_REQUEST_VERB verb)
 
     return NULL;
 }
+
+#if defined(KVS_USE_OPENSSL)

```

Comment:
I strongly discourage usage of conditional compilations in this manner. 

Recommend using a separate header that hides the #ifdef under a macro and use those macros in the single code body. 

You only need to abstract two API calls. sha256 and hmac.